Branch-SDK/src/main/java/io/branch/referral/validators/IntegrationValidatorDialog.java

```diff
@@ -0,0 +1,109 @@
+    IntegrationValidatorDialogRowItem test8RowItem;
+
+    Button exportLogsButton;
+
+    public IntegrationValidatorDialog(final Context context) {
+        super(context);
+        requestWindowFeature(Window.FEATURE_NO_TITLE);
+        this.setContentView(R.layout.dialog_integration_validator);
+        TextView sdkVersionTextView = findViewById(R.id.sdk_version);
+        sdkVersionTextView.setText("SDK Version: " + Branch.getSdkVersionNumber());
+
+        test1RowItem = findViewById(R.id.test_1_auto_instance_validator_row);
+        test2RowItem = findViewById(R.id.test_2_verify_branch_keys);
+        test3RowItem = findViewById(R.id.test_3_verify_package_name);
+        test4RowItem = findViewById(R.id.test_4_verify_uri_scheme);
+        test5RowItem = findViewById(R.id.test_5_verify_app_links);
+        test6RowItem = findViewById(R.id.test_6_verify_custom_domain);
+        test7RowItem = findViewById(R.id.test_7_domain_intent_filters);
+        test8RowItem = findViewById(R.id.test_8_alternate_domain_intent_filters);
+
+        exportLogsButton = findViewById(R.id.export_logs_button);
+
+        exportLogsButton.setOnClickListener(view -> {
+            shareLogsAsText(context);
+        });
+    }
+
+    public void setTestResult(int testNumber, String name, boolean didTestPass, String detailsMessage, String moreInfoLink) {
+        switch (testNumber) {
+            case 1:
+                setResult(test1RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 2:
+                setResult(test2RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 3:
+                setResult(test3RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 4:
+                setResult(test4RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 5:
+                setResult(test5RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 6:
+                setResult(test6RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 7:
+                setResult(test7RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+            case 8:
+                setResult(test8RowItem, name, didTestPass, detailsMessage, moreInfoLink);
+                break;
+        }
+    }
+
+    private void setResult(IntegrationValidatorDialogRowItem rowItem, String name, boolean didTestPass, String detailsMessage, String moreInfoLink) {
+        rowItem.SetTitleText(name);
+        rowItem.SetTestResult(didTestPass);
+        rowItem.SetDetailsMessage(detailsMessage);
+        rowItem.SetMoreInfoLink(moreInfoLink);
+    }
+
+    private void shareLogsAsText(Context context) {
+        try {
+            Process process = Runtime.getRuntime().exec("logcat -d BranchSDK:V *:S");
```

Comment:
I would recommend instead to use the interface to define your own behavior with the logs that we generate in memory directly rather than reaching into the system.

See: https://github.com/BranchMetrics/android-branch-deep-linking-attribution/blob/master/Branch-SDK/src/main/java/io/branch/interfaces/IBranchLoggingCallbacks.java

https://github.com/BranchMetrics/android-branch-deep-linking-attribution/blob/master/Branch-SDK-TestBed/src/main/java/io/branch/branchandroidtestbed/CustomBranchApp.java#L17

For each log generated, you can aggregate and store it in memory in a StringBuilder for example, and then share that text.

Reply:
@gdeluna-branch good idea. I did as you requested. However, it doesn't seem like the BranchSDK_Doctor logs get sent in that callback. I do see other Branch logs in there though. Is there some additional modification I need to make to BranchLogger.kt or another class? Latest commit pushed with the changes. 

Here is an example of the Branch logs I get back from that callback: 

> !SDK-VERSION-STRING!:io.branch.sdk.android:library:5.12.3handleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977handleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 needs a sessionprocessNextQueueItem handleNewRequestQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977onActivityResumed, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onIntentReady io.branch.branchandroidtestbed.MainActivity@94c0343 removing INTENT_PENDING_WAIT_LOCKRead params uri: null bypassCurrentActivityIntentState: false intent state: READYextractExternalUriAndIntentExtras null io.branch.branchandroidtestbed.MainActivity@94c0343isIntentParamsAlreadyConsumed falseAttempt to invoke virtual method 'java.lang.String android.net.Uri.toString()' on a null object referenceextractBranchLinkFromIntentExtra io.branch.branchandroidtestbed.MainActivity@94c0343processNextQueueItem onIntentReadyQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977initializing session on user's behalf (onActivityResumed called but SESSION_STATE = UNINITIALISED)Beginning session initializationSession uri is nullCallback is nullIs auto init trueWill ignore intent nullIs reinitializing falseisInstantDeepLinkPossible falseCreating io.branch.referral.ServerRequestRegisterOpen@1644e1b from init on thread maininitializeSession io.branch.referral.ServerRequestRegisterOpen@1644e1b delay 0Warning: You are using your test app's Branch Key. Remember to change it to live Branch Key during deployment.Intent: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 cmp=io.branch.branchandroidtestbed/.MainActivity } forceBranchSession: false initState: UNINITIALISEDregisterAppInit io.branch.referral.ServerRequestRegisterOpen@1644e1bOrdering init callsQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> Moving io.branch.referral.ServerRequestRegisterOpen@1644e1b  to front of the queue or behind network-in-progress requestFinished ordering init callsQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> initTasks io.branch.referral.ServerRequestRegisterOpen@1644e1b ignoreWaitLocks falseprocessNextQueueItem registerAppInitQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@1644e1bonActivityPaused, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onActivityStopped, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onActivityStarted, activity = io.branch.branchandroidtestbed.MainActivity@94c0343!SDK-VERSION-STRING!:io.branch.sdk.android:library:5.12.3handleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85handleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 needs a sessionprocessNextQueueItem handleNewRequestQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@1644e1bonActivityResumed, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onIntentReady io.branch.branchandroidtestbed.MainActivity@94c0343 removing INTENT_PENDING_WAIT_LOCKRead params uri: null bypassCurrentActivityIntentState: false intent state: READYextractExternalUriAndIntentExtras null io.branch.branchandroidtestbed.MainActivity@94c0343isIntentParamsAlreadyConsumed falseAttempt to invoke virtual method 'java.lang.String android.net.Uri.toString()' on a null object referenceextractBranchLinkFromIntentExtra io.branch.branchandroidtestbed.MainActivity@94c0343processNextQueueItem onIntentReadyQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@1644e1binitializing session on user's behalf (onActivityResumed called but SESSION_STATE = UNINITIALISED)Beginning session initializationSession uri is nullCallback is nullIs auto init trueWill ignore intent nullIs reinitializing falseisInstantDeepLinkPossible falseCreating io.branch.referral.ServerRequestRegisterOpen@ba3cf0 from init on thread maininitializeSession io.branch.referral.ServerRequestRegisterOpen@ba3cf0 delay 0Warning: You are using your test app's Branch Key. Remember to change it to live Branch Key during deployment.Intent: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 cmp=io.branch.branchandroidtestbed/.MainActivity } forceBranchSession: false initState: UNINITIALISEDregisterAppInit io.branch.referral.ServerRequestRegisterOpen@ba3cf0Ordering init callsQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> Moving io.branch.referral.ServerRequestRegisterOpen@ba3cf0  to front of the queue or behind network-in-progress requestFinished ordering init callsQueue is: io.branch.referral.ServerRequestRegisterOpen@ba3cf0 with locks []
> io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> initTasks io.branch.referral.ServerRequestRegisterOpen@ba3cf0 ignoreWaitLocks falseprocessNextQueueItem registerAppInitQueue is: io.branch.referral.ServerRequestRegisterOpen@ba3cf0 with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@ba3cf0processNextQueueItem onAdsParamsFetchFinishedQueue is: io.branch.referral.ServerRequestRegisterOpen@ba3cf0 with locks []
> io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@ba3cf0executeTimedBranchPostTask io.branch.referral.ServerRequestRegisterOpen@ba3cf0callback to be returned nullonPreExecute io.branch.referral.ServerRequestRegisterOpen@ba3cf0processNextQueueItem onAdsParamsFetchFinishedQueue is: io.branch.referral.ServerRequestRegisterOpen@ba3cf0 with locks []
> io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks [SDK_INIT_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks [SDK_INIT_WAIT_LOCK]
> Current referringURLQueryParameters: {}Beginning rest post for io.branch.referral.ServerRequestRegisterOpen@ba3cf0posting to https://api2.branch.io/v1/openPost value = {"randomized_device_token":"1378371027381669962","randomized_bundle_token":"1253180467713556816","hardware_id":"50dba345-99b6-49c0-86f8-96e634f43fef","is_hardware_id_real":false,"anon_id":"b501dde7-215c-477f-af7e-14f7fd174326","brand":"samsung","model":"SM-G991U1","screen_dpi":480,"screen_height":2176,"screen_width":1080,"wifi":true,"ui_mode":"UI_MODE_TYPE_NORMAL","os":"Android","os_version":34,"country":"US","language":"en","local_ip":"192.168.1.164","cpu_type":"aarch64","build":"UP1A.231005.007.G991U1UESDGXJ1","locale":"en_US","connection_type":"wifi","os_version_android":"14","debug":false,"partner_data":{"fb":{"em":"194b86d986ad041666822dad7602f1a7bac1d9e286273e86141666ffb4b1909b","ph":"7cd2dfb0b893767d1b0a4cacd831f6f9cf0ace6b1fbfc616c313c672d08f6196"}},"app_version":"5.12.3","initial_referrer":"android-app:\/\/com.android.shell","update":1,"latest_install_time":1729867484905,"latest_update_time":1729867484905,"first_install_time":1729867484905,"previous_update_time":1729867484905,"environment":"FULL_APP","metadata":{},"app_store":"PlayStore","advertising_ids":{"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"},"lat_val":0,"google_advertising_id":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd","sdk":"android5.12.3","branch_key":"key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc"}lastResponseMessage OKBranch Networking Success
> URL: https://api2.branch.io/v1/open
> Response Code: 200
> Response Message: OK
> Retry number: 0
> requestId: 8af209fa-7b16-4e00-979f-47a7d65a161a-2024102515
> Object: io.branch.referral.network.BranchRemoteInterfaceUrlConnection@9b658f5Server returned: [8af209fa-7b16-4e00-979f-47a7d65a161a-2024102515] Status: [200]; Data: {"data":"{\"+clicked_branch_link\":false,\"+is_first_session\":false}","identity":"testDevID","link":"[https://bnctestbed.test-app.link?%24randomized_bundle_token=1253180467713556816](https://bnctestbed.test-app.link/?%24randomized_bundle_token=1253180467713556816)","randomized_bundle_token":"1253180467713556816","randomized_device_token":"1378371027381669962","session_id":"1378380910961055923"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@43cb68a io.branch.referral.ServerResponse@633f4fbonRequestSuccess io.branch.referral.ServerResponse@633f4fbDoes not have Clicked_Branch_Link or Clicked_Branch_Link is false, returningpostInitClear io.branch.referral.PrefHelper@de5f918 can clear init data falseprocessNextQueueItem unlockSDKInitWaitLockQueue is: io.branch.referral.ServerRequestRegisterOpen@ba3cf0 with locks []
> io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> onRequestSucceeded io.branch.referral.ServerRequestRegisterOpen@ba3cf0 io.branch.referral.ServerResponse@633f4fb on callback nullonInitSessionCompleted on thread mainprocessNextQueueItem onPostExecuteInnerQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@1644e1bexecuteTimedBranchPostTask io.branch.referral.ServerRequestRegisterOpen@1644e1bcallback to be returned nullonPreExecute io.branch.referral.ServerRequestRegisterOpen@1644e1bBegin getUserAgentAsync Thread[DefaultDispatcher-worker-2,5,main]Current referringURLQueryParameters: {}Beginning rest post for io.branch.referral.ServerRequestRegisterOpen@1644e1bposting to https://api2.branch.io/v1/openPost value = {"randomized_device_token":"1378371027381669962","randomized_bundle_token":"1253180467713556816","hardware_id":"50dba345-99b6-49c0-86f8-96e634f43fef","is_hardware_id_real":false,"anon_id":"b501dde7-215c-477f-af7e-14f7fd174326","brand":"samsung","model":"SM-G991U1","screen_dpi":480,"screen_height":2176,"screen_width":1080,"wifi":true,"ui_mode":"UI_MODE_TYPE_NORMAL","os":"Android","os_version":34,"country":"US","language":"en","local_ip":"192.168.1.164","cpu_type":"aarch64","build":"UP1A.231005.007.G991U1UESDGXJ1","locale":"en_US","connection_type":"wifi","os_version_android":"14","debug":false,"partner_data":{"fb":{"em":"194b86d986ad041666822dad7602f1a7bac1d9e286273e86141666ffb4b1909b","ph":"7cd2dfb0b893767d1b0a4cacd831f6f9cf0ace6b1fbfc616c313c672d08f6196"}},"app_version":"5.12.3","initial_referrer":"android-app:\/\/com.android.shell","update":1,"latest_install_time":1729867484905,"latest_update_time":1729867484905,"first_install_time":1729867484905,"previous_update_time":1729867484905,"environment":"FULL_APP","metadata":{},"app_store":"PlayStore","advertising_ids":{"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"},"lat_val":0,"google_advertising_id":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd","instrumentation":{"v1\/open-brtt":"302"},"sdk":"android5.12.3","branch_key":"key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc"}End getUserAgentAsync Thread[DefaultDispatcher-worker-2,5,main] Mozilla/5.0 (Linux; Android 14; SM-G991U1 Build/UP1A.231005.007; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/130.0.6723.58 Mobile Safari/537.36onInitSessionCompleted resumeWith userAgent Mozilla/5.0 (Linux; Android 14; SM-G991U1 Build/UP1A.231005.007; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/130.0.6723.58 Mobile Safari/537.36 on thread DefaultDispatcher-worker-2processNextQueueItem getUserAgentAsync resumeWithQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> lastResponseMessage OKBranch Networking Success
> URL: https://api2.branch.io/v1/open
> Response Code: 200
> Response Message: OK
> Retry number: 0
> requestId: f5d75018-2444-431b-a4ec-3cdae4142ebf-2024102515
> Object: io.branch.referral.network.BranchRemoteInterfaceUrlConnection@9b658f5Server returned: [f5d75018-2444-431b-a4ec-3cdae4142ebf-2024102515] Status: [200]; Data: {"data":"{\"+clicked_branch_link\":false,\"+is_first_session\":false}","identity":"testDevID","link":"[https://bnctestbed.test-app.link?%24randomized_bundle_token=1253180467713556816](https://bnctestbed.test-app.link/?%24randomized_bundle_token=1253180467713556816)","randomized_bundle_token":"1253180467713556816","randomized_device_token":"1378371027381669962","session_id":"1378380911493926370"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@25b2ca1 io.branch.referral.ServerResponse@df13ec6onRequestSuccess io.branch.referral.ServerResponse@df13ec6Does not have Clicked_Branch_Link or Clicked_Branch_Link is false, returningpostInitClear io.branch.referral.PrefHelper@de5f918 can clear init data trueprocessNextQueueItem unlockSDKInitWaitLockQueue is: io.branch.referral.ServerRequestRegisterOpen@1644e1b with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> onRequestSucceeded io.branch.referral.ServerRequestRegisterOpen@1644e1b io.branch.referral.ServerResponse@df13ec6 on callback nullDeferring userAgent string call for sync retrievalonInitSessionCompleted on thread mainprocessNextQueueItem onPostExecuteInnerQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977executeTimedBranchPostTask io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977onPreExecute io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977getting [https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata={}&advertising_ids={](https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata=%7B%7D&advertising_ids=%7B)"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"}&hardware_id=50dba345-99b6-49c0-86f8-96e634f43fef&is_hardware_id_real=false&lat_val=0&google_advertising_id=9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd&sdk=android5.12.3&branch_key=key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJcServer returned: [9028da44fec641f48629289a5e1d928b-2024102515] Status: [200]; Data: {"android_uri_scheme":"branchtest://","android_package_name":"io.branch.branchandroiddemo","short_url_domain":"","default_short_url_domain":"bnctestbed.test-app.link","alternate_short_url_domain":"bnctestbed-alternate.test-app.link","ios_uri_scheme":"branchtest://","ios_team_id":"R63EM248DP","ios_bundle_id":"io.branch.sdk.Branch-TestBed"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@4672d4e io.branch.referral.ServerResponse@8103e6fonRequestSuccess io.branch.referral.ServerResponse@8103e6fhandleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@aca977cprocessNextQueueItem handleNewRequestQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@e9ef977 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@aca977c with locks []
> processNextQueueItem onPostExecuteInnerQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85 with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@aca977c with locks []
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85executeTimedBranchPostTask io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85onPreExecute io.branch.referral.validators.ServerRequestGetAppConfig@a9edd85getting [https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata={}&advertising_ids={](https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata=%7B%7D&advertising_ids=%7B)"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"}&hardware_id=50dba345-99b6-49c0-86f8-96e634f43fef&is_hardware_id_real=false&lat_val=0&google_advertising_id=9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd&sdk=android5.12.3&branch_key=key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJcServer returned: [1ce921c761244fe88cb32e4e70689110-2024102515] Status: [200]; Data: {"android_uri_scheme":"branchtest://","android_package_name":"io.branch.branchandroiddemo","short_url_domain":"","default_short_url_domain":"bnctestbed.test-app.link","alternate_short_url_domain":"bnctestbed-alternate.test-app.link","ios_uri_scheme":"branchtest://","ios_team_id":"R63EM248DP","ios_bundle_id":"io.branch.sdk.Branch-TestBed"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@45f83dc io.branch.referral.ServerResponse@f005ce5onRequestSuccess io.branch.referral.ServerResponse@f005ce5processNextQueueItem onPostExecuteInnerQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@aca977c with locks []
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@aca977cexecuteTimedBranchPostTask io.branch.referral.validators.ServerRequestGetAppConfig@aca977conPreExecute io.branch.referral.validators.ServerRequestGetAppConfig@aca977cgetting [https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata={}&advertising_ids={](https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata=%7B%7D&advertising_ids=%7B)"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"}&hardware_id=50dba345-99b6-49c0-86f8-96e634f43fef&is_hardware_id_real=false&lat_val=0&google_advertising_id=9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd&sdk=android5.12.3&branch_key=key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJcServer returned: [6ad1c13c2aaa45d3b1b736508363d488-2024102515] Status: [200]; Data: {"android_uri_scheme":"branchtest://","android_package_name":"io.branch.branchandroiddemo","short_url_domain":"","default_short_url_domain":"bnctestbed.test-app.link","alternate_short_url_domain":"bnctestbed-alternate.test-app.link","ios_uri_scheme":"branchtest://","ios_team_id":"R63EM248DP","ios_bundle_id":"io.branch.sdk.Branch-TestBed"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@5cdf26b io.branch.referral.ServerResponse@24f01c8onRequestSuccess io.branch.referral.ServerResponse@24f01c8processNextQueueItem onPostExecuteInnerQueue is: onActivityPaused, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onActivityStopped, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onActivityStarted, activity = io.branch.branchandroidtestbed.MainActivity@94c0343!SDK-VERSION-STRING!:io.branch.sdk.android:library:5.12.3handleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@4e1722fhandleNewRequest io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f needs a sessionprocessNextQueueItem handleNewRequestQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@4e1722fonActivityResumed, activity = io.branch.branchandroidtestbed.MainActivity@94c0343onIntentReady io.branch.branchandroidtestbed.MainActivity@94c0343 removing INTENT_PENDING_WAIT_LOCKRead params uri: null bypassCurrentActivityIntentState: false intent state: READYextractExternalUriAndIntentExtras null io.branch.branchandroidtestbed.MainActivity@94c0343isIntentParamsAlreadyConsumed falseAttempt to invoke virtual method 'java.lang.String android.net.Uri.toString()' on a null object referenceextractBranchLinkFromIntentExtra io.branch.branchandroidtestbed.MainActivity@94c0343processNextQueueItem onIntentReadyQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@4e1722finitializing session on user's behalf (onActivityResumed called but SESSION_STATE = UNINITIALISED)Beginning session initializationSession uri is nullCallback is nullIs auto init trueWill ignore intent nullIs reinitializing falseisInstantDeepLinkPossible falseCreating io.branch.referral.ServerRequestRegisterOpen@ce80bdb from init on thread maininitializeSession io.branch.referral.ServerRequestRegisterOpen@ce80bdb delay 0Warning: You are using your test app's Branch Key. Remember to change it to live Branch Key during deployment.Intent: Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 cmp=io.branch.branchandroidtestbed/.MainActivity } forceBranchSession: false initState: UNINITIALISEDregisterAppInit io.branch.referral.ServerRequestRegisterOpen@ce80bdbOrdering init callsQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks [SDK_INIT_WAIT_LOCK]
> Moving io.branch.referral.ServerRequestRegisterOpen@ce80bdb  to front of the queue or behind network-in-progress requestFinished ordering init callsQueue is: io.branch.referral.ServerRequestRegisterOpen@ce80bdb with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks [SDK_INIT_WAIT_LOCK]
> initTasks io.branch.referral.ServerRequestRegisterOpen@ce80bdb ignoreWaitLocks falseprocessNextQueueItem registerAppInitQueue is: io.branch.referral.ServerRequestRegisterOpen@ce80bdb with locks [GAID_FETCH_WAIT_LOCK]
> io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@ce80bdbprocessNextQueueItem onAdsParamsFetchFinishedQueue is: io.branch.referral.ServerRequestRegisterOpen@ce80bdb with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks [SDK_INIT_WAIT_LOCK]
> processNextQueueItem, req io.branch.referral.ServerRequestRegisterOpen@ce80bdbexecuteTimedBranchPostTask io.branch.referral.ServerRequestRegisterOpen@ce80bdbcallback to be returned nullonPreExecute io.branch.referral.ServerRequestRegisterOpen@ce80bdbCurrent referringURLQueryParameters: {}Beginning rest post for io.branch.referral.ServerRequestRegisterOpen@ce80bdbposting to https://api2.branch.io/v1/openPost value = {"randomized_device_token":"1378371027381669962","randomized_bundle_token":"1253180467713556816","hardware_id":"50dba345-99b6-49c0-86f8-96e634f43fef","is_hardware_id_real":false,"anon_id":"b501dde7-215c-477f-af7e-14f7fd174326","brand":"samsung","model":"SM-G991U1","screen_dpi":480,"screen_height":2176,"screen_width":1080,"wifi":true,"ui_mode":"UI_MODE_TYPE_NORMAL","os":"Android","os_version":34,"country":"US","language":"en","local_ip":"192.168.1.164","cpu_type":"aarch64","build":"UP1A.231005.007.G991U1UESDGXJ1","locale":"en_US","connection_type":"wifi","os_version_android":"14","debug":false,"partner_data":{"fb":{"em":"194b86d986ad041666822dad7602f1a7bac1d9e286273e86141666ffb4b1909b","ph":"7cd2dfb0b893767d1b0a4cacd831f6f9cf0ace6b1fbfc616c313c672d08f6196"}},"app_version":"5.12.3","initial_referrer":"android-app:\/\/com.android.shell","update":1,"latest_install_time":1729867484905,"latest_update_time":1729867484905,"first_install_time":1729867484905,"previous_update_time":1729867484905,"environment":"FULL_APP","metadata":{},"app_store":"PlayStore","advertising_ids":{"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"},"lat_val":0,"google_advertising_id":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd","instrumentation":{"v1\/open-brtt":"122","v1\/app-link-settings-brtt":"199"},"sdk":"android5.12.3","branch_key":"key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc"}lastResponseMessage OKBranch Networking Success
> URL: https://api2.branch.io/v1/open
> Response Code: 200
> Response Message: OK
> Retry number: 0
> requestId: 32fa0d99-816c-4399-a568-d81664c7c130-2024102515
> Object: io.branch.referral.network.BranchRemoteInterfaceUrlConnection@9b658f5Server returned: [32fa0d99-816c-4399-a568-d81664c7c130-2024102515] Status: [200]; Data: {"data":"{\"+clicked_branch_link\":false,\"+is_first_session\":false}","identity":"testDevID","link":"[https://bnctestbed.test-app.link?%24randomized_bundle_token=1253180467713556816](https://bnctestbed.test-app.link/?%24randomized_bundle_token=1253180467713556816)","randomized_bundle_token":"1253180467713556816","randomized_device_token":"1378371027381669962","session_id":"1378381000056461774"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@8649f42 io.branch.referral.ServerResponse@a6a1b53onRequestSuccess io.branch.referral.ServerResponse@a6a1b53Does not have Clicked_Branch_Link or Clicked_Branch_Link is false, returningpostInitClear io.branch.referral.PrefHelper@de5f918 can clear init data trueprocessNextQueueItem unlockSDKInitWaitLockQueue is: io.branch.referral.ServerRequestRegisterOpen@ce80bdb with locks []
> io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks []
> onRequestSucceeded io.branch.referral.ServerRequestRegisterOpen@ce80bdb io.branch.referral.ServerResponse@a6a1b53 on callback nullDeferring userAgent string call for sync retrievalonInitSessionCompleted on thread mainprocessNextQueueItem onPostExecuteInnerQueue is: io.branch.referral.validators.ServerRequestGetAppConfig@4e1722f with locks []
> processNextQueueItem, req io.branch.referral.validators.ServerRequestGetAppConfig@4e1722fexecuteTimedBranchPostTask io.branch.referral.validators.ServerRequestGetAppConfig@4e1722fonPreExecute io.branch.referral.validators.ServerRequestGetAppConfig@4e1722fgetting [https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata={}&advertising_ids={](https://api2.branch.io/v1/app-link-settings/key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJc?metadata=%7B%7D&advertising_ids=%7B)"aaid":"9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd"}&hardware_id=50dba345-99b6-49c0-86f8-96e634f43fef&is_hardware_id_real=false&lat_val=0&google_advertising_id=9ef34cfa-d7b9-41a5-9baa-5cd8cb2da0bd&sdk=android5.12.3&branch_key=key_test_hdcBLUy1xZ1JD0tKg7qrLcgirFmPPVJcServer returned: [f2b31a2ce2eb4561b496c446253ea39e-2024102515] Status: [200]; Data: {"android_uri_scheme":"branchtest://","android_package_name":"io.branch.branchandroiddemo","short_url_domain":"","default_short_url_domain":"bnctestbed.test-app.link","alternate_short_url_domain":"bnctestbed-alternate.test-app.link","ios_uri_scheme":"branchtest://","ios_team_id":"R63EM248DP","ios_bundle_id":"io.branch.sdk.Branch-TestBed"}onPostExecuteInner io.branch.referral.ServerRequestQueue$BranchPostTask@8f119af io.branch.referral.ServerResponse@d571bconRequestSuccess io.branch.referral.ServerResponse@d571bcprocessNextQueueItem onPostExecuteInnerQueue is:

Reply:
Thinking about this more, I think that as is it would help support. The Branch doctor output would probably just be for the customer, so no need for us to add that into the shared text IMO.